server/client.go

```diff
@@ -281,8 +281,12 @@ func (c *client) readLoop() {
 		c.cache.subs = 0
 
 		if err := c.parse(b[:n]); err != nil {
-			// handled inline
-			if err != ErrMaxPayload && err != ErrAuthorization {
+			// If client connection has been closed, simply return,
+			// otherwise report a generic parsing error.
+			c.mu.Lock()
+			closed := c.nc == nil
+			c.mu.Unlock()
+			if !closed {
```

Comment:
I have restored the behavior. The intent was that I did not want to print the "Error reading from client: ..." error, but at the same time I forgot to print a specific one ;-)
The reason for the MaxPayload and Authorization check is that these two print their own error, send the error back to client and close the connection, so they don't need this generic error message.